server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -574,7 +576,11 @@ void moveShards() {
 
             ShardRouting shardRouting = it.next();
 
-            if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getShardPool(shardRouting, allocation))) {
+            // if routing pool is remote capable and
+            // - if feature flag is not enabled, skip, or
+            // - if feature flag is enabled and shard is not partial, skip
+            if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getShardPool(shardRouting, allocation))
+                && (!FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX) || (!isPartialShard(shardRouting, allocation)))) {
```

Comment:
Small nit:
We could rewrite `!FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX) || (!isPartialShard(shardRouting, allocation))` as `!(FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX) && isPartialShard(shardRouting, allocation))`.

IMO that's clearer to read as code but maybe a little hard to describe as code comment so will leave that up to you.

Reply:
sure, will update this